controller/controller.go

```diff
@@ -49,55 +54,40 @@ type Config struct {
 	// list of resources being executed for the received runtime object can be
 	// versioned and different resources can be executed depending on the runtime
 	// object being reconciled.
-	ResourceSets []*ResourceSet
-	// RESTClient needs to be configured with a serializer capable of serializing
-	// and deserializing the object which is watched by the informer. Otherwise
-	// deserialization will fail when trying to add a finalizer.
-	//
-	// For standard k8s object this is going to be e.g.
-	//
-	// 		k8sClient.CoreV1().RESTClient()
-	//
-	// For CRs of giantswarm this is going to be e.g.
-	//
-	// 		g8sClient.CoreV1alpha1().RESTClient()
-	//
-	RESTClient rest.Interface
+	ResourceSets         []*ResourceSet
+	RuntimeObjectFactory func() pkgruntime.Object
```

Comment:
I think this name is more go-native. Also `obj := c.newRuntimeObjectFunc()` reads much better than `obj := c.runtimeObjectFactory()` IMO.

```suggestion
	NewRuntimeObjectFunc func() pkgruntime.Object
```

`NewRuntimeObject` would also work for me. I don't know how much value is in adding `Func` suffix. For me it allows to distinguish between value-function and method.